question_nine.py

```diff
@@ -0,0 +1,24 @@
+from portion import closed
+from datetime import datetime
+
+start_datestr = '2024-05-07 1'
+end_datestr='2024-05-08 22'
+
+date_format = "%Y-%m-%d %H"
+
+start_date = datetime.strptime(start_datestr, date_format)
+end_date = datetime.strptime(end_datestr, date_format)
+
+keep_intervals = closed(start_date.hour, end_date.hour) 
+#print(keep_intervals)
+day_difference = (end_date - start_date).days
+#print(day_difference)
+
+day_difference=day_difference*18
+hours=0
+
+for i in range(keep_intervals.lower,keep_intervals.upper):
+    hours+=1
```

Comment:
Same here, we can't add the hours directly to 1.

What if I give you dates like `2023-11-10T10:10:23` and `2023-11-15T15:10:59`. I still want exact difference no round off values. That's why we need to use portion for each calculation. Thank you!